### PR TITLE
Adding new HMM-based conservation pipeline.

### PR DESCRIPTION
**Pull request by davidjakubec**
Accelerated conservation calculations using HMMER. New pipeline files added to conservation_hmm/.

### CODE REVIEW

**Comment by skodapetr** [comment]
Some general comments:
 * 80 characters per line
 * Use double quotes
 * Add README.md file into the directory with some description (installation step, or link to repository where the instalation steps are ... )

**Comment by davidjakubec** [comment]
The changes are in! I have:
 - formatted `conservation_hmm.py` with black for ultra-orthodox PEP8 compliance;
 - added README.md with installation and other instructions;
 - renamed tests/ to examples/ and moved database/ there to improve clarity.

File: conservation_hmm/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+`conservation_hmm.py` is a Python script which can be used to calculate per-position information content (IC) values for amino acid residues in a FASTA file. It is inspired by the [INTAA-conservation](https://github.com/davidjakubec/INTAA-conservation) pipeline utilized by the [Amino Acid Interactions (INTAA) web server](https://bioinfo.uochb.cas.cz/INTAA/). Unlike INTAA-conservation, `conservation_hmm.py` skips parsing the PDB structure and works with the single-sequence FASTA files directly. For further information on the pipeline's logic, please refer to the [INTAA manual](https://ip-78-128-251-188.flt.cloud.muni.cz/energy/doc/manual2.html#Calculation_of_information_content).
+
+## Installation
+
+`conservation_hmm.py` is a transferable, executable script which does not require any installation. It does not depend on any other Python packages or files. For other dependencies, please see the section below.
+
+### Requirements
+
+The [HMMER](http://hmmer.org/) software package, including the Easel tools, must be installed and the programs must be available in `$PATH`. All development and testing is done using HMMER 3.3.2.
```
Comment (skodapetr) [inline conservation_hmm/README.md:13]: I would merge this to installation and add example on how to install on Debian.

File: conservation_hmm/README.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+The script `examples/database/download_Swiss-Prot.sh` can be used to download an example sequence database.
```
Comment (skodapetr) [inline conservation_hmm/README.md:15]: I would remove the script and rather put the command here directly. The script is small and I would not expect user to actually use it. Keep in mind that it now becomes part of a single package not a standalone application.

File: conservation_hmm/README.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+## Usage
+
+Use
+```
+conservation_hmm.py FASTA_file database_file working_directory target_file
```
Comment (skodapetr) [inline conservation_hmm/README.md:21]: All the arguments are positional, can't we use arguments like ```--fasta-file```, ```--working-directory```.
Also I would move this into the python script and here show example of commands to run, including the download command.

As user can then type ```conservation_hmm.py -h``` to get the help. That seems better to me, as I can get the help when and where I need it, in the terminal.

File: conservation_hmm/README.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+ - `database_file` is a sequence database used to construct the multiple sequence alignments (MSAs) based on which the IC is later calculated. It is usually a regular file in a multiple-sequence FASTA format.
+ - `working_directory` is a directory in which the MSAs and other temporary files are stored. **It must end with your OS's path delimiting character** \[*i.e.*, a slash \("/"\) on UNIX systems\] **in the script's arguments.** The temporary files in `working_directory` are **not** cleaned up after the script terminates.
+ - `target_file` is the primary output file containing the per-position IC values. For its format and additional information, please see the section below.
+
+### Output
+
+When the script terminates, the `target_file` will contain a list of tab-separated triples \(index, IC, amino\_acid\_residue\) for the amino acid residues in the `FASTA_file`, where index is simply a number starting from zero \(0\) for the first residue. One triple is provided per line.
```
Comment (skodapetr) [inline conservation_hmm/README.md:31]: Again I would perhaps merge this into the use-case example with some example. 

The idea is that here in this document, the user can find a list of commands to run. So I can try it easily on my own.

File: conservation_hmm/conservation_hmm.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python3
+
+from sys import argv
+from subprocess import DEVNULL, run
+
+
+def generate_MSA(FASTA_file, database_file, working_directory):
+    run(
+        "phmmer -o /dev/null -A {}{}.sto {} {}".format(
```
Comment (skodapetr) [inline conservation_hmm/conservation_hmm.py:9]: The Python we are using is 3.7, so there is already support for f-strings. I would highly recommend using them.

**Review by skodapetr — CHANGES_REQUESTED** [review]

**Pull request merged**